lib/protoparser/prometheus/parser.go

```diff
@@ -34,27 +42,32 @@ func (rs *Rows) Reset() {
 	rs.tagsPool = rs.tagsPool[:0]
 }
 
-// Unmarshal unmarshals Prometheus exposition text rows from s.
+// Unmarshal unmarshals Prometheus exposition (protobuf and text formats) rows from s.
 //
 // See https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md#text-format-details
 //
 // s shouldn't be modified while rs is in use.
-func (rs *Rows) Unmarshal(s string) {
-	rs.UnmarshalWithErrLogger(s, stdErrLogger)
+func (rs *Rows) Unmarshal(s string, contentType string) {
```

Comment:
I'd suggest do not change signature of exist functions and provide new method instead.

something like:

`func (rs *Rows) UnmarshalProto(src []byte) {`